gtsam/sfm/SfmData.h

```diff
@@ -61,6 +82,31 @@ struct SfmData {
   /// The camera pose at frame index `idx`
   SfmCamera camera(size_t idx) const { return cameras[idx]; }
 
+  /**
+   * @brief Create projection factors using keys i and P(j)
+   *
+   * @param model a noise model for projection errors
+   * @return NonlinearFactorGraph
+   */
+  NonlinearFactorGraph generalSfmFactors(
+      const SharedNoiseModel& model = noiseModel::Isotropic::Sigma(2,
+                                                                   1.0)) const;
+
+  /**
+   * @brief Create factor graph with projection factors and gauge fix.
+   *
+   * Note: pose keys are simply integer indices, points use Symbol('p', j).
+   *
+   * @param model a noise model for projection errors
+   * @param fixedCamera which camera to fix, if any (use boost::none if none)
+   * @param fixedPoint which point to fix, if any (use boost::none if none)
+   * @return NonlinearFactorGraph
+   */
+  NonlinearFactorGraph sfmFactorGraph(
+      const SharedNoiseModel& model = noiseModel::Isotropic::Sigma(2, 1.0),
+      boost::optional<size_t> fixedCamera = 0,
+      boost::optional<size_t> fixedPoint = 0) const;
```

Comment:
Nit, feel free to ignore if merging is urgent: 
I think it is unclear what the difference between generalSfmFactors and sfmFactorGraph is to a user from the name itself. Maybe sfmFactorGraphWithPriors is a better name?

Reply:
generalSfmFactors is sorta internal, so maybe I will move that to a different {} group